data/ignition/files/usr/local/bin/set-node-zero.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+source common.sh
+
+HOST=$(get_host)
+echo Using hostname ${HOST} 1>&2
+
+if [[ ${HOST} == {{.NodeZeroIP}} ]] ;then
```

Comment:
I think it would be reasonable to start assisted-service if the NodeZeroIP appears on any interface (i.e. not just if it's the one with the default route).